modules/programs/mangohud.nix

```
@@ -0,0 +1,100 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.mangohud;
+
+  renderOption = option:
+    rec {
+      int = toString option;
+      float = int;
+      path = int;
+      bool = "false";
+      string = option;
+      list = concatStringsSep "," option;
+    }.${builtins.typeOf option};
+
+  renderLine = k: v: (if isBool v && v then k else "${k}=${renderOption v}");
+  renderSettings = attrs:
+    strings.concatStringsSep "\n" (attrsets.mapAttrsToList renderLine attrs)
+    + "\n";
+
+in {
+  options = {
+    programs.mangohud = {
+      enable = mkEnableOption "Mangohud";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.mangohud;
+        defaultText = literalExample "pkgs.mangohud";
+        description = "The Mangohud package to install.";
+      };
+
+      setEnvVars = mkOption {
+        type = types.bool;
+        default = false;
+        description =
+          "Sets the environment variable so that mangohud is started on any application that supports it";
+      };
+
+      settings = mkOption {
+        type = with types;
+          attrsOf (oneOf [ bool int float str path (listOf str) ]);
+        default = { };
+        example = literalExample ''
+          {
+            output_folder = ~/Documents/mangohud/;
+            full = true;
+          }
+        '';
+        description = ''
+          Configuration written to
+          <filename>~/.config/MangoHud/MangoHud.conf</filename>. See
+          <link xlink:href="https://github.com/flightlessmango/MangoHud/blob/master/bin/MangoHud.conf"/>
+          for the default configuration.
+        '';
+      };
+      settingsPerApplication = mkOption {
```

Separate option definitions by an empty line
```suggestion

      settingsPerApplication = mkOption {
```